store/organizations.js

```diff
@@ -109,15 +187,52 @@ const $actions = {
 
     if (organizations) commit(UPDATE_CACHE, organizations.records);
 
-    return organizations;
+    return _.cloneDeep(organizations);
+  },
+
+  async getOrganization({ commit, getters }, { id, allowReadFromCache }) {
+    let organization = null;
+
+    if (allowReadFromCache && getters.isOrganizationCached(id)) {
+      organization = getters.getOrganization(id);
+    } else {
+      organization = await this.$kronosApi.getOrganization(id);
+      if (organization) commit(UPDATE_CACHE, [ organization ]);
+    }
+
+    return _.cloneDeep(organization);
+  },
+
+  async createOrganization({ commit }, organization) {
+    const result = await this.$kronosApi.createOrganization(organization);
+
+    if (result) commit(UPDATE_CACHE, [ result ]);
+
+    return _.cloneDeep(result);
   },
 
-  async getOrganizationById({ commit }, id) {
-    const organization = await this.$kronosApi.getOrganization(id);
+  async updateOrganization({ commit }, organization) {
+    const result = await this.$kronosApi.updateOrganization(organization);
+
+    if (result) commit(UPDATE_CACHE, [ result ]);
+
+    return _.cloneDeep(result);
+  },
 
-    if (organization) commit(UPDATE_CACHE, [ organization ]);
+  async deleteOrganization({ commit }, organizationId) {
+    await this.$kronosApi.deleteOrganization(organizationId);
 
-    return organization;
+    commit(REMOVE_CACHE, organizationId);
+  },
+
+  async getOrganizationContacts({ commit, getters }, { id, allowReadFromCache }) {
+    const contacts = allowReadFromCache && getters.isOrganizationContactsCached(id)
+      ? getters.getOrganizationContacts(id)
+      : await this.$kronosApi.getOrganizationContacts(id);
+
+    commit(UPDATE_ORGANIZATION_CONTACTS_CACHE, { organizationId: id, contacts });
```

Comment:
Only update cache if read from api
```suggestion
    let contacts = null
    
    if(allowReadFromCache && getters.isOrganizationContactsCached(id) {
      contacts = getters.getOrganizationContacts(id)
    } else {
      contacts = await this.$kronosApi.getOrganizationContacts(id);
      commit(UPDATE_ORGANIZATION_CONTACTS_CACHE, { organizationId: id, contacts });
    }
```

Reply:
DONE! changes done as suggested